R/module_nested_tabs.R

```diff
@@ -178,56 +163,93 @@ srv_nested_tabs.teal_modules <- function(id, datasets, modules, is_module_specif
   })
 }
 
-#' @rdname module_nested_tabs
+#' @rdname module_teal_module
 #' @export
-srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specific = TRUE,
+srv_teal_module.teal_module <- function(id,
+                                        data_rv,
+                                        datasets,
+                                        modules,
                                         reporter = teal.reporter::Reporter$new()) {
-  checkmate::assert_class(datasets, "FilteredData")
-  logger::log_trace("srv_nested_tabs.teal_module initializing the module: { deparse1(modules$label) }.")
-
+  logger::log_trace("srv_teal_module.teal_module initializing the module: { deparse1(modules$label) }.")
   moduleServer(id = id, module = function(input, output, session) {
-    if (!is.null(modules$datanames) && is_module_specific) {
-      datasets$srv_filter_panel("module_filter_panel")
+    active_datanames <- reactive({
+      req(data_rv())
+      if (is.null(modules$datanames) || identical(modules$datanames, "all")) {
+        teal_data_datanames(data_rv())
+      } else {
+        include_parent_datanames(
+          modules$datanames,
+          teal.data::join_keys(data_rv())
+        )
+      }
+    })
+    if (is.null(datasets)) {
+      datasets <- eventReactive(data_rv(), {
+        logger::log_trace("srv_teal_module@1 initializing module-specific FilteredData")
+
+        # Otherwise, FilteredData will be created in the modules' scope later
+        progress_data <- Progress$new(
+          max = length(unlist(module_labels(modules)))
+        )
+        on.exit(progress_data$close())
+        progress_data$set(message = "Preparing data filtering", detail = "0%")
+        filtered_data <- teal_data_to_filtered_data(data_rv(), datanames = active_datanames())
+        filtered_data
```

Comment:
who is handling this progress bar? shall we pass it to the `teal_data_to_filtered_data()` or maybe this function would create, handle and close it by itself